src/NuGetForUnity/Editor/NugetHelper.cs

```diff
@@ -72,6 +72,11 @@ public static class NugetHelper
         /// </summary>
         private static readonly Dictionary<string, NugetPackage> installedPackages = new Dictionary<string, NugetPackage>();
 
+        /// <summary>
+        ///     The dictionary of currently selected installed <see cref="NugetPackage" />s keyed off of their ID string.
+        /// </summary>
+        private static readonly Dictionary<string, NugetPackage> selectedInstalledPackages = new Dictionary<string, NugetPackage>();
```

Comment:
I know the code of `NuGetForUnity` is not the most clean but pleas try not to pollute the `NugetHelper` class. As far as I see you only need this inside the UI (`NugetWindow`) so I don't see a reason to place this inside the "global" `NugetHelper` class. Also you are "duplicating" the information you have a collection of selected + a bool on each package. Duplicating is normally not a good practice as it can get out of sync. I think we should only use one collection and no bool as you are storing a state of a single UI so it should not interfere with other code. I think you already have the issue that you don't know if the user selected a package for uninstall or for updating as you store both in the same location.